src/libmongoc/src/mongoc/mongoc-util.h

```diff
@@ -0,0 +1,34 @@
+#include "mongoc-prelude.h"
+
+#ifndef MONGOC_UTIL_H
+#define MONGOC_UTIL_H
+
+#include <bson/bson.h>
+
+#include "mongoc-macros.h"
+
+BSON_BEGIN_DECLS
+
+/**
+ * mongoc_usleep_func_t:
+ * @usec: Number of microseconds to sleep for.
+ * @user_data: User data provided to mongoc_usleep_set_impl().
+ */
+typedef void (*mongoc_usleep_func_t) (int64_t usec, void *user_data);
+
+/**
+ * mongoc_usleep_set_impl:
+ * @usleep_func: A function to perform microsecond sleep.
+ *
+ * Sets the function to be called to perform sleep.
+ * Not thread-safe.
```

Comment:
```suggestion
 * Not thread-safe.
 * Providing a `usleep_func` that does not sleep (e.g. coroutine suspension) is
 * not supported. Doing so is at the user's own risk.
```

At present, suspending threads before blocking work is not a supported use-case of libmongoc. This note is intended to advise users of the risk.

Reply:
done